### PR TITLE
Add pprof

### PR DESCRIPTION
**Pull request by lujiajing1126**
This PR intends to add `pprof` for runtime diagnosis

### CODE REVIEW

File: banyand/prof/pprof.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+type pprofService struct {
+	listenAddr string
+	stopCh     chan struct{}
+	l          *logger.Logger
+}
+
+func (p *pprofService) FlagSet() *run.FlagSet {
+	flagSet := run.NewFlagSet("prof")
+	flagSet.StringVar(&p.listenAddr, "pprof-listener-addr", "0.0.0.0:6060", "listen addr for pprof")
```
Comment (wu-sheng) [inline banyand/prof/pprof.go:48]: Exporting to all network access? This seems a security leak.

Comment (lujiajing1126) [reply banyand/prof/pprof.go]: Fixed

File: banyand/prof/pprof.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+type pprofService struct {
+	listenAddr string
+	stopCh     chan struct{}
+	l          *logger.Logger
+}
+
+func (p *pprofService) FlagSet() *run.FlagSet {
+	flagSet := run.NewFlagSet("prof")
+	flagSet.StringVar(&p.listenAddr, "pprof-listener-addr", "127.0.0.1:6060", "listen addr for pprof")
```
Comment (wu-sheng) [inline banyand/prof/pprof.go:48]: @hanahmily need your suggestion here. Hard code seems not a good practice. WDYT?

Comment (lujiajing1126) [reply banyand/prof/pprof.go:48]: > @hanahmily need your suggestion here. Hard code seems not a good practice. WDYT?

`127.0.0.1:6060` is the default value for the flag which can be changed with command line option `--prof-listener-addr` at runtime.

Comment (wu-sheng) [reply banyand/prof/pprof.go:48]: Then where should be documented? I feel this kind of thing will be missed from time to time.

Comment (lujiajing1126) [reply banyand/prof/pprof.go:48]: > Then where should be documented? I feel this kind of thing will be missed from time to time.

Currently the flag system designed by @hanahmily is self-described,

```
$ ./banyand/build/bin/banyand-server standalone --help
2022-01-28T10:08:32+08:00 | DEBUG | ***register flags**** module:STANDALONE name:METADATA registered:1 total:4
2022-01-28T10:08:32+08:00 | DEBUG | ***register flags**** module:STANDALONE name:STREAM registered:2 total:4
2022-01-28T10:08:32+08:00 | DEBUG | ***register flags**** module:STANDALONE name:GRPC registered:3 total:4
2022-01-28T10:08:32+08:00 | DEBUG | ***register flags**** module:STANDALONE name:PPROF-SERVICE registered:4 total:4
Run as the standalone mode

Usage:
   standalone [flags]

Flags:
      --addr string                  The address of banyand listens (default ":17912")
      --cert-file string             The TLS cert file
  -h, --help                         help for standalone
      --key-file string              The TLS key file
      --logging.env string           the logging (default "dev")
      --logging.level string         the level of logging (default "debug")
      --max-recv-msg-size int        The size of max receiving message (default 10485760)
      --metadata-root-path string    the root path of metadata (default "/tmp")
  -n, --name string                  name of this service (default "standalone")
      --pprof-listener-addr string   listen addr for pprof (default "127.0.0.1:6060")
      --root-path string             the root path of database (default "/tmp")
      --show-rungroup-units          show rungroup units
      --tls                          Connection uses TLS if true, else plain TCP
  -v, --version                      version for standalone
```

all flags registered will be automatically shown in the help messages. I am not sure if this is what you want.

Comment (wu-sheng) [reply banyand/prof/pprof.go:48]: OK, thanks for sharing. I am good now.

**Review by hanahmily — APPROVED** [review]
LGTM

**Pull request merged**